src/components/BuildDid/BuildDid.tsx

```diff
@@ -0,0 +1,310 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  Fragment,
+  MouseEvent,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import { parse } from '@kiltprotocol/asset-did';
+
+import * as styles from './BuildDid.module.css';
+
+import { useBooleanState } from '../../utilities/useBooleanState';
+import { isAssetDidUri } from '../../utilities/isAssetDidUri';
+import { isOpenSeaUrl, parseOpenSeaUrl } from '../../utilities/openSea';
+import { AssetDidElements } from '../AssetDid/AssetDid';
+
+interface NFT {
+  contract_address: string;
+  token_id: string;
+  name?: string;
+  cached_file_url?: string;
+  cached_animation_url?: string;
+}
+
+type SupportedMetaMaskChainId = '0x1' | '0x5' | '0x89';
+type SupportedCaip2ChainId = 'eip155:1' | 'eip155:5' | 'eip155:137';
+type SupportedApiChainId = 'ethereum' | 'goerli' | 'polygon';
+
+const supportedCaip2Chains: Record<
+  SupportedMetaMaskChainId,
+  SupportedCaip2ChainId
+> = {
+  '0x1': 'eip155:1',
+  '0x5': 'eip155:5',
+  '0x89': 'eip155:137',
+};
+
+const supportedApiChains: Record<SupportedCaip2ChainId, SupportedApiChainId> = {
+  'eip155:1': 'ethereum',
+  'eip155:5': 'goerli',
+  'eip155:137': 'polygon',
+};
+
+function isSupportedChain(
+  chainId: string,
+): chainId is SupportedMetaMaskChainId {
+  return ['0x1', '0x5', '0x89'].includes(chainId);
+}
+
+function useMetamask(reset: () => void) {
+  const hasMetaMask = useBooleanState();
+
+  const [chainId, setChainId] = useState<SupportedCaip2ChainId>();
+  const [account, setAccount] = useState<string>();
+
+  const handleAccountsChanged = useCallback(
+    (accounts: string) => {
```

Comment:
Would that be more appropriate?
```suggestion
    (accounts: string[]) => {
```